rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -309,4 +300,72 @@ void test_extractRedeemScriptFromInput_migration_multiple_inputs() {
+    }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionIsSegwit_shouldThrowIllegalArgumentException() {
+        // arrange
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        transaction.addInput(BitcoinTestUtils.createHash(1), 0, new Script(new byte[]{}));
+        TransactionWitness transactionWitness = new TransactionWitness(1);
+        transactionWitness.setPush(0, new byte[]{});
+        transaction.setWitness(0, transactionWitness);
+
+        // act & assert
+        IllegalArgumentException exception = Assertions.assertThrows(IllegalArgumentException.class,
+            () -> BitcoinUtils.removeSignaturesFromTransactionWithInputsWithP2shMultiSigInputScript(transaction));
+        Assertions.assertEquals("Removing signatures from SegWit transactions is not allowed.", exception.getMessage());
+    }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionInputsDoNotHaveP2shMultiSigInputScript_shouldThrowIllegalArgumentException() {
+        // arrange
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        BtcECKey pubKey = new BtcECKey();
+        Script p2pkhScriptSig = ScriptBuilder.createInputScript(null, pubKey);
+        transaction.addInput(BitcoinTestUtils.createHash(1), 0, p2pkhScriptSig);
+
+        // act & assert
+        IllegalArgumentException exception = Assertions.assertThrows(IllegalArgumentException.class,
+            () -> BitcoinUtils.removeSignaturesFromTransactionWithInputsWithP2shMultiSigInputScript(transaction));
+        Assertions.assertEquals("Cannot remove signatures from transaction inputs that do not have p2sh multisig input script.", exception.getMessage());
+    }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionIsLegacyAndInputsHaveP2shMultiSigInputScript_shouldRemoveSignatures() {
+        // arrange
+        Federation federation = new P2shErpFederationBuilder().build();
+        Script scriptSig = federation.getP2SHScript().createEmptyInputScript(null, federation.getRedeemScript());
+
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        transaction.addInput(BitcoinTestUtils.createHash(1), 0, scriptSig);
+        transaction.addInput(BitcoinTestUtils.createHash(2), 0, scriptSig);
+        transaction.addOutput(Coin.COIN, destinationAddress);
+        Sha256Hash transactionWithoutSignaturesHash = transaction.getHash();
+
+        List<BtcECKey> keysToSign =
+            BitcoinTestUtils.getBtcEcKeysFromSeeds(new String[]{"member01", "member02", "member03", "member04", "member05"}, true); // using private keys from federation declared above
+        List<TransactionInput> inputs = transaction.getInputs();
+        for (TransactionInput input : inputs) {
+            BitcoinTestUtils.signTransactionInputFromMultiSigWithKeys(transaction, inputs.indexOf(input), keysToSign);
```

Comment:
No big deal, but since you need the input index sometimes it's simpler to just the classic `for` 

```
for (i=0; i<inputs.lenght;i++)
```
That way you already have the index

Reply:
yeah i know, but i thought it was cleaner this way